File: src/agent/pkg/api/models/policy.go
package models

// PolicyRequest represents a policy creation/update request
type PolicyRequest struct {
	RuleID   uint32 `json:"rule_id" binding:"required"`
	SrcIP    string `json:"src_ip" binding:"required"`
	DstIP    string `json:"dst_ip" binding:"required"`
	SrcPort  uint16 `json:"src_port"`
	DstPort  uint16 `json:"dst_port"`
	Protocol string `json:"protocol" binding:"required,oneof=tcp udp icmp any"`
	Action   string `json:"action" binding:"required,oneof=allow deny log"`
	Priority uint16 `json:"priority"`
}

// PolicyResponse represents a policy in API responses
type PolicyResponse struct {
	RuleID   uint32 `json:"rule_id"`
	SrcIP    string `json:"src_ip"`
	DstIP    string `json:"dst_ip"`
	SrcPort  uint16 `json:"src_port"`
	DstPort  uint16 `json:"dst_port"`
	Protocol string `json:"protocol"`
	Action   string `json:"action"`
	Priority uint16 `json:"priority"`
}

// PolicyListResponse represents a list of policies
type PolicyListResponse struct {
	Policies []PolicyResponse `json:"policies"`
	Count    int              `json:"count"`
}

File: web/src/types/policy.ts
// Network policy rule
export interface Policy {
  ruleId: number
  srcIp: string // CIDR notation
  dstIp: string // CIDR notation
  srcPort: number // 0 for any
  dstPort: number // 0 for any
  protocol: 'tcp' | 'udp' | 'icmp' | 'any'
  action: 'allow' | 'deny' | 'log'
  priority: number
  enabled?: boolean
  description?: string
  createdAt?: string
  updatedAt?: string
}

// Policy statistics
export interface PolicyStats {
  ruleId: number
  hitCount: number
  lastHit?: string
}

// Policy list response
export interface PolicyListResponse {
  policies: Policy[]
  total: number
}

// Policy validation result
export interface PolicyValidation {
  valid: boolean
  errors?: string[]
  warnings?: string[]
}
